salt/fileserver/s3fs.py

```diff
@@ -591,7 +591,9 @@ def _prune_deleted_files(metadata):
                     roots.add(root)
 
             for meta in env_data:
-                cached_files.add(meta["Key"])
+                for bucket in meta.keys():
+                    for obj in meta[bucket]:
```

Comment:
To iterate over items (keyword, value pairs) in a dict, use `.items()`, but this isn't using the key so just use `.values()`

```python
for objects in meta.values():
    for obj in objects:
```